### PR TITLE
New printing API

### PR DESCRIPTION
**Pull request by hazeled**
Adds "TCOD_printf", "TCOD_printn", and "TCOD_vprintf" (Along with a PrintParams struct) to console_printing. Also fixes some typos

### CODE REVIEW

File: src/libtcod/console_printing.c [hunk 1]
```diff
@@ -1517,4 +1517,44 @@ TCOD_Error TCOD_console_printf_frame(
...
+  int err = TCOD_console_print_internal(console, params.x, params.y, params.width, params.height, params.flag, params.alignment, strbuf, true, false);
+  free(strbuf);
+  return err;
+}
+TCOD_PUBLIC int TCOD_vprintf(
+    TCOD_Console* __restrict console,
+    TCOD_PrintParams params,
+    const char* fmt,
+    va_list args) {
+  int err = TCOD_console_print_internal(console, params.x, params.y, params.width, params.height, params.flag, params.alignment, TCOD_console_vsprint(fmt, args), true, false);
```
[the remark this answers is not in the document]
Comment (HexDecimal) [reply src/libtcod/console_printing.c:1557]: Also `TCOD_console_vsprint` is not as reentrant as it looks, and should be avoided.  `vprintf_internal_` handles this case.

**Review by HexDecimal — CHANGES_REQUESTED** [review]
I didn't realize how non-obvious it was that everything above the `// New UTF-8 parser.` line is something to be avoided.

It's very unfortunate that you performed this work on your `main` branch instead of a separate branch.  I can't safely push changes to the PR in this state.  You will also be unable to make a 2nd PR until this is fixed.

I don't like the unversioned print functions.  They should at least have something such as `rgb` as in `TCOD_printn_rgb` and `TCOD_PrintParamsRGB` to at least define that this version is using the RGB color types.

Tests will need to be added to `tests/test_printing.cpp` to satisfy Codecov.

If you're ever done working on this PR then it's possible for me to merge this now.  I'll have an easier time resolving these issues since I'm more familiar with the library and the issues are relatively small.

**Comment by hazeled** [comment]
Ahh, my bad. I haven't worked on this library in a while. I'll go ahead and fix those issues, thank you for pointing them out

**Comment by hazeled** [comment]
Also, I somehow didn't realize I was pushing to the main branch of my fork. I can move it to a different fork with a different branch, if you think that'd work best

**Comment by HexDecimal** [comment]
Move this to a topic branch and then I'll start pushing changes to that branch.  Most of the issues are small and should be easy for me to fix.

**Pull request closed without merging**